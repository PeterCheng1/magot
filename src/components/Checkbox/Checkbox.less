@import '../../variable.less';

.@{prefix}-checkbox {
  position: relative;
  display: inline-block;
  cursor: pointer;
  .box-sizing();
  .box();

  &:hover {
    .@{prefix}-checkbox-control {
      &::before {
        border-color: @checkbox-check-border-color;
      }
    }
  }
}

.@{prefix}-checkbox-native-control {
  font-size: 0;
  opacity: 0.0001;
  cursor: pointer;
  .overlay();

  &:checked {
    & + .@{prefix}-checkbox-control {
      &::before {
        border-color: @checkbox-check-border-color;
        background-color: @checkbox-check-background-color;
      }
      &::after {
        content: '';
        position: absolute;
        top: @checkbox-checker-top;
        left: @checkbox-checker-left;
        width: @checkbox-checker-width;
        height: @checkbox-checker-height;
        transform: rotate(45deg);
        border: @checkbox-checker-border-width solid
          @checkbox-checker-border-color;
        border-top: 0;
        border-left: 0;
        transition: all ease-out 200ms 100ms;
      }
    }
  }
}

.@{prefix}-checkbox-control {
  display: inline-block;
  position: relative;
  vertical-align: middle;

  &::before {
    content: '';
    display: block;
    border-width: @checkbox-border-width;
    border-style: @checkbox-border-style;
    border-color: @checkbox-uncheck-border-color;
    background-color: @checkbox-uncheck-background-color;
    border-radius: @checkbox-border-radius;
    transition: all ease-out 300ms;
    .square(@checkbox-control-size);
    .box-sizing();
  }
}

.@{prefix}-checkbox-text {
  color: @checkbox-text-color;
  font-size: @checkbox-text-font-size;
  padding-left: @checkbox-text-gutter;
  vertical-align: middle;
}

.@{prefix}-checkbox-disabled {
  opacity: 0.35;
  pointer-events: none;
  cursor: not-allowed;
}
