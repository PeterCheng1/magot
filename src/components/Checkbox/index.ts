export { default, CheckboxProps, CheckboxValue } from './Checkbox';
export { CheckboxGroupProps, CheckboxOption } from '../CheckboxGroup';
