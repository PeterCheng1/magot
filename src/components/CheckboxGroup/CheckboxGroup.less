@import '../../variable.less';

.mgt-checkbox-group {
  position: relative;
}
