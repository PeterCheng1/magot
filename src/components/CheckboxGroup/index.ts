export { default, CheckboxGroupProps, CheckboxOption } from './CheckboxGroup';
