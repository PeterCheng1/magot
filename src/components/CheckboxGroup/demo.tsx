import React from 'react';
import { Link } from 'react-router-dom';
import Checkbox, { CheckboxOption } from '../Checkbox';

function CheckboxGroupDemo() {
  const options: CheckboxOption[] = [
    {
      label: 'Option1',
      value: 'option1',
      checked: true,
    },
    {
      label: 'Option2',
      value: 'option2',
    },
    {
      label: 'Option3',
      value: 'option3',
      disabled: true,
    },
  ];
  return (
    <>
      <Link to="/" className="demo-goback">
        返回
      </Link>
      <div className="demo-box">
        <Checkbox.Group options={options} onChange={console.log} />
        <br />
        <Checkbox.Group options={options} disabled={true} />
      </div>
      <div className="demo-box">
        <Checkbox.Group onChange={console.log}>
          <Checkbox value="option1" checked={true}>
            Option1
          </Checkbox>
          <Checkbox value="option2">Option2</Checkbox>
          <Checkbox value="option3" disabled={true}>
            Option3
          </Checkbox>
        </Checkbox.Group>
        <br />
        <Checkbox.Group disabled={true}>
          <Checkbox value="option1" checked={true}>
            Option1
          </Checkbox>
          <Checkbox value="option2">Option2</Checkbox>
          <Checkbox value="option3" disabled={true}>
            Option3
          </Checkbox>
        </Checkbox.Group>
      </div>
    </>
  );
}

export default CheckboxGroupDemo;
