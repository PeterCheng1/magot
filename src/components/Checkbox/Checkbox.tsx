import React from 'react';

import CheckboxGroup from '../CheckboxGroup';
import * as component from '../component';

import './Checkbox.less';

export interface CheckboxProps
  extends component.FormComponent<HTMLInputElement>,
    component.NestedComponent {
  /**
   * 当前是否选中，用于受控组件，默认不选中
   * @see https://reactjs.org/docs/forms.html?#controlled-components
   * @default false
   */
  checked?: boolean;

  /**
   * 默认是否选中，用于非受控组件
   * @see https://reactjs.org/docs/uncontrolled-components.html
   * @default false
   */
  defaultChecked?: boolean;

  defaultValue?: string;

  /**
   * 选中或反选时的回调函数
   */
  onChange?: (
    checked: boolean,
    value: React.ReactText | undefined,
    e: React.ChangeEvent<HTMLInputElement>
  ) => void;
}

const defaultProps: Partial<CheckboxProps> = {
  ...component.getDefaultDisabledProps(),
  defaultChecked: false,
};

function Checkbox(props: CheckboxProps) {
  const isControlled = 'checked' in props;
  const { children, className, style, onChange, ...formProps } = props;

  const type = 'checkbox';
  const prefix = component.getComponentPrefix(type);
  const cls = component.getComponentClasses(type, props);

  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    onChange && onChange(e.target.checked, props.value, e);
  };

  return (
    <label className={cls} style={style}>
      <input
        {...formProps}
        type="checkbox"
        className={prefix + '-native-control'}
        checked={isControlled ? props.checked : undefined}
        defaultChecked={!isControlled ? props.defaultChecked : undefined}
        onChange={handleChange}
      />
      <span className={prefix + '-control'} />
      {children && <span className={prefix + '-text'}>{children}</span>}
    </label>
  );
}

Checkbox.defaultProps = defaultProps;
Checkbox.Group = CheckboxGroup;

export default Checkbox;
