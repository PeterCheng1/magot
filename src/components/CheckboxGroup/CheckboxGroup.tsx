import React from 'react';

import './CheckboxGroup.less';

import * as component from '../component';

export interface CheckboxGroupProps extends component.ComponentBase {}

const defaultProps: Partial<CheckboxGroupProps> = {};

function CheckboxGroup(props: CheckboxGroupProps) {
  const cls = component.getComponentClasses('checkbox-group', props);
  return <div className={cls} />;
}

CheckboxGroup.defaultProps = defaultProps;

export default CheckboxGroup;
