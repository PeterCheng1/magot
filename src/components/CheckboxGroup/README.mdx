---
name: CheckboxGroup
route: /components/CheckboxGroup
menu: Components
---

import { Playground, PropsTable } from 'docz';
import CheckboxGroup from './CheckboxGroup';

# CheckboxGroup

## Usage

## props

<PropsTable of={CheckboxGroup} />
