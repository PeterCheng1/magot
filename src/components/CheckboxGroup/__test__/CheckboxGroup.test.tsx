import React from 'react';
import { render, mount } from 'enzyme';
import Checkbox from '../../Checkbox';
import CheckboxGroup, { CheckboxOption } from '../CheckboxGroup';
import { injectStyleTestSuites } from '../../../../test/shared.test';

describe('CheckboxGroup', () => {
  it('should renders dom correctly', () => {
    const wrapper = render(
      <CheckboxGroup>
        <Checkbox value="option1">Option1</Checkbox>
        <Checkbox value="option2">Option2</Checkbox>
        <Checkbox value="option3">Option3</Checkbox>
      </CheckboxGroup>
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.children()).toHaveLength(3);
  });

  it('should renders by options', () => {
    const options: CheckboxOption[] = [
      {
        label: 'Option1',
        value: 'option1',
      },
      {
        label: 'Option2',
        value: 'option2',
      },
      {
        label: 'Option3',
        value: 'option3',
      },
    ];
    const wrapper = render(<CheckboxGroup options={options} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.children()).toHaveLength(3);
  });

  it('should renders empty children', () => {
    const wrapper = render(<CheckboxGroup />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.children()).toHaveLength(0);
  });

  it('should renders `value` prop when passed in', () => {
    const wrapper = render(
      <CheckboxGroup value={['option1']}>
        <Checkbox value="option1">Option1</Checkbox>
        <Checkbox value="option2">Option2</Checkbox>
        <Checkbox value="option3">Option3</Checkbox>
      </CheckboxGroup>
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('input[checked]').val()).toEqual('option1');
  });

  it('should renders `disabled` prop when passed in', () => {
    const wrapper = render(
      <CheckboxGroup disabled={true}>
        <Checkbox value="option1">Option1</Checkbox>
        <Checkbox value="option2">Option2</Checkbox>
        <Checkbox value="option3">Option3</Checkbox>
      </CheckboxGroup>
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.hasClass('mgt-checkbox-group-disabled')).toBe(true);
    expect(wrapper.find('input[disabled]')).toHaveLength(3);
  });

  it('should renders `name` prop when passed in', () => {
    const wrapper = render(
      <CheckboxGroup name="group">
        <Checkbox value="option1">Native Group1</Checkbox>
        <Checkbox value="option2">Native Group2</Checkbox>
        <Checkbox value="option3">Native Group3</Checkbox>
      </CheckboxGroup>
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('input[name="group"]')).toHaveLength(3);
  });

  it('should simulates change event', () => {
    const handleChange = jest.fn();
    const wrapper = mount(
      <CheckboxGroup value={['option1']} onChange={handleChange}>
        <Checkbox value="option1">Option1</Checkbox>
        <Checkbox value="option2">Option2</Checkbox>
        <Checkbox value="option3">Option3</Checkbox>
      </CheckboxGroup>
    );
    const eventData = { target: { checked: true } };
    wrapper.find('input[value="option2"]').simulate('change', eventData);
    expect(handleChange).toHaveBeenCalledWith(['option1', 'option2']);
  });

  injectStyleTestSuites(CheckboxGroup);
});
