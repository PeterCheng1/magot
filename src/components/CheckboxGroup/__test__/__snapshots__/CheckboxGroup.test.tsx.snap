// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxGroup should renders \`disabled\` prop when passed in 1`] = `
<div
  class="mgt-checkbox-group mgt-checkbox-group-disabled"
>
  <label
    class="mgt-checkbox mgt-checkbox-disabled"
  >
    <input
      class="mgt-checkbox-native-control"
      disabled=""
      type="checkbox"
      value="option1"
    />
    <span
      class="mgt-checkbox-control"
    />
    <span
      class="mgt-checkbox-text"
    >
      Option1
    </span>
  </label>
  <label
    class="mgt-checkbox mgt-checkbox-disabled"
  >
    <input
      class="mgt-checkbox-native-control"
      disabled=""
      type="checkbox"
      value="option2"
    />
    <span
      class="mgt-checkbox-control"
    />
    <span
      class="mgt-checkbox-text"
    >
      Option2
    </span>
  </label>
  <label
    class="mgt-checkbox mgt-checkbox-disabled"
  >
    <input
      class="mgt-checkbox-native-control"
      disabled=""
      type="checkbox"
      value="option3"
    />
    <span
      class="mgt-checkbox-control"
    />
    <span
      class="mgt-checkbox-text"
    >
      Option3
    </span>
  </label>
</div>
`;

exports[`CheckboxGroup should renders \`name\` prop when passed in 1`] = `
<div
  class="mgt-checkbox-group"
>
  <label
    class="mgt-checkbox"
  >
    <input
      class="mgt-checkbox-native-control"
      name="group"
      type="checkbox"
      value="option1"
    />
    <span
      class="mgt-checkbox-control"
    />
    <span
      class="mgt-checkbox-text"
    >
      Native Group1
    </span>
  </label>
  <label
    class="mgt-checkbox"
  >
    <input
      class="mgt-checkbox-native-control"
      name="group"
      type="checkbox"
      value="option2"
    />
    <span
      class="mgt-checkbox-control"
    />
    <span
      class="mgt-checkbox-text"
    >
      Native Group2
    </span>
  </label>
  <label
    class="mgt-checkbox"
  >
    <input
      class="mgt-checkbox-native-control"
      name="group"
      type="checkbox"
      value="option3"
    />
    <span
      class="mgt-checkbox-control"
    />
    <span
      class="mgt-checkbox-text"
    >
      Native Group3
    </span>
  </label>
</div>
`;

exports[`CheckboxGroup should renders \`value\` prop when passed in 1`] = `
<div
  class="mgt-checkbox-group"
>
  <label
    class="mgt-checkbox"
  >
    <input
      checked=""
      class="mgt-checkbox-native-control"
      type="checkbox"
      value="option1"
    />
    <span
      class="mgt-checkbox-control"
    />
    <span
      class="mgt-checkbox-text"
    >
      Option1
    </span>
  </label>
  <label
    class="mgt-checkbox"
  >
    <input
      class="mgt-checkbox-native-control"
      type="checkbox"
      value="option2"
    />
    <span
      class="mgt-checkbox-control"
    />
    <span
      class="mgt-checkbox-text"
    >
      Option2
    </span>
  </label>
  <label
    class="mgt-checkbox"
  >
    <input
      class="mgt-checkbox-native-control"
      type="checkbox"
      value="option3"
    />
    <span
      class="mgt-checkbox-control"
    />
    <span
      class="mgt-checkbox-text"
    >
      Option3
    </span>
  </label>
</div>
`;

exports[`CheckboxGroup should renders by options 1`] = `
<div
  class="mgt-checkbox-group"
>
  <label
    class="mgt-checkbox"
  >
    <input
      class="mgt-checkbox-native-control"
      type="checkbox"
      value="option1"
    />
    <span
      class="mgt-checkbox-control"
    />
    <span
      class="mgt-checkbox-text"
    >
      Option1
    </span>
  </label>
  <label
    class="mgt-checkbox"
  >
    <input
      class="mgt-checkbox-native-control"
      type="checkbox"
      value="option2"
    />
    <span
      class="mgt-checkbox-control"
    />
    <span
      class="mgt-checkbox-text"
    >
      Option2
    </span>
  </label>
  <label
    class="mgt-checkbox"
  >
    <input
      class="mgt-checkbox-native-control"
      type="checkbox"
      value="option3"
    />
    <span
      class="mgt-checkbox-control"
    />
    <span
      class="mgt-checkbox-text"
    >
      Option3
    </span>
  </label>
</div>
`;

exports[`CheckboxGroup should renders custom className 1`] = `
<div
  class="mgt-checkbox-group jest-test-class"
/>
`;

exports[`CheckboxGroup should renders custom styles 1`] = `
<div
  class="mgt-checkbox-group"
  style="color:#333;background-color:#f0f0f0"
/>
`;

exports[`CheckboxGroup should renders dom correctly 1`] = `
<div
  class="mgt-checkbox-group"
>
  <label
    class="mgt-checkbox"
  >
    <input
      class="mgt-checkbox-native-control"
      type="checkbox"
      value="option1"
    />
    <span
      class="mgt-checkbox-control"
    />
    <span
      class="mgt-checkbox-text"
    >
      Option1
    </span>
  </label>
  <label
    class="mgt-checkbox"
  >
    <input
      class="mgt-checkbox-native-control"
      type="checkbox"
      value="option2"
    />
    <span
      class="mgt-checkbox-control"
    />
    <span
      class="mgt-checkbox-text"
    >
      Option2
    </span>
  </label>
  <label
    class="mgt-checkbox"
  >
    <input
      class="mgt-checkbox-native-control"
      type="checkbox"
      value="option3"
    />
    <span
      class="mgt-checkbox-control"
    />
    <span
      class="mgt-checkbox-text"
    >
      Option3
    </span>
  </label>
</div>
`;

exports[`CheckboxGroup should renders empty children 1`] = `
<div
  class="mgt-checkbox-group"
/>
`;
